sdk/src/wallet/account/operations/transaction/high_level/send_amount.rs

```diff
@@ -52,20 +94,80 @@ impl AccountHandle {
     pub async fn prepare_send_amount(
         &self,
         addresses_with_amount: Vec<AddressWithAmount>,
-        options: Option<TransactionOptions>,
+        options: impl Into<Option<TransactionOptions>>,
     ) -> crate::wallet::Result<PreparedTransactionData> {
         log::debug!("[TRANSACTION] prepare_send_amount");
-        let mut outputs = Vec::new();
+        let options = options.into();
+        let rent_structure = self.client.get_rent_structure().await?;
         let token_supply = self.client.get_token_supply().await?;
 
-        for address_with_amount in addresses_with_amount {
-            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address_with_amount.address)?;
+        let account_addresses = self.addresses().await?;
+        let default_return_address = account_addresses.first().ok_or(Error::FailedToGetRemainder)?;
+
+        let local_time = self.client.get_time_checked().await?;
+
+        let mut outputs = Vec::new();
+        for AddressWithAmount {
+            address,
+            amount,
+            return_address,
+            expiration,
+        } in addresses_with_amount
+        {
+            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address)?;
+            let return_address = return_address
+                .map(|address| Ok::<_, Error>(Address::try_from_bech32_with_hrp(address)?.0))
+                .transpose()?
+                .unwrap_or(default_return_address.address.inner);
             self.client.bech32_hrp_matches(&bech32_hrp).await?;
-            outputs.push(
-                BasicOutputBuilder::new_with_amount(address_with_amount.amount)?
-                    .add_unlock_condition(AddressUnlockCondition::new(address))
-                    .finish_output(token_supply)?,
-            )
+
+            // Get the minimum required amount for an output assuming it does not need a storage deposit.
+            let output = BasicOutputBuilder::new_with_amount(amount)?
+                .add_unlock_condition(AddressUnlockCondition::new(address))
+                .finish_output(token_supply)?;
+            let rent_cost = output.rent_cost(&rent_structure);
+
+            if amount >= rent_cost {
+                outputs.push(output)
+            } else {
+                let expiration_time = expiration.map_or(local_time + DEFAULT_EXPIRATION_TIME, |expiration_time| {
+                    local_time + expiration_time
+                });
+
+                // Since it does need a storage deposit, calculate how much that should be
+                let storage_deposit_amount = minimum_storage_deposit_basic_native_tokens(
+                    &rent_structure,
+                    &address,
+                    &return_address,
+                    None,
+                    token_supply,
+                )?;
+
+                if !options.as_ref().map(|o| o.allow_micro_amount).unwrap_or_default() {
+                    return Err(Error::InsufficientFunds {
```

Comment:
Is this variant really always correct 🤔 ?

Reply:
Well, technically I think that error means there's insufficient funds in the account, but it was pretty close and the error isn't *wrong* so 🤷